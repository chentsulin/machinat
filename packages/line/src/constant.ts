export const CHANNEL_REQUEST_GETTER = Symbol(
  'channel_request_getter.line.machinat'
);
export const BULK_REQUEST_GETTER = Symbol('bulk_request_getter.line.machinat');

/** @internal */
export const LINE = 'line' as const;

/** @internal */
export const PATH_REPLY = 'v2/bot/message/reply';

/** @internal */
export const PATH_PUSH = 'v2/bot/message/push';

/** @internal */
export const PATH_MULTICAST = 'v2/bot/message/multicast';

/** @internal */
export const PATH_RICHMENU = 'v2/bot/richmenu';

export enum LiffContextOs {
  Ios,
  Android,
  Web,
}
