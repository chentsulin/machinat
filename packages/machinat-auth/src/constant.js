// @flow
import { namedInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const AUTH_SERVER_AUTHORIZERS_I = namedInterface<
  ServerAuthorizer<any, any>[]
>({
  name: 'ServerAuthorizersList',
  multi: true,
});

export const AUTH_MODULE_CONFIGS_I = namedInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
