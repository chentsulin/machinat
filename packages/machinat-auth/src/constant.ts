import { makeInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const SERVER_AUTHORIZERS_I = makeInterface<ServerAuthorizer<any, any>[]>(
  { name: 'ServerAuthorizersList', multi: true }
);

export const MODULE_CONFIGS_I = makeInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
