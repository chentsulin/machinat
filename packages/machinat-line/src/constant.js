// @flow
export const LINE = 'line';
export const LINE_NAMESPACE = 'Line';

export const LINE_NATIVE_TYPE = Symbol.for('machinat.line.native');

export const ENTRY_GETTER = Symbol.for('machinat.line.segment.entry_getter');

export const PATH_REPLY = 'v2/bot/message/reply';
export const PATH_PUSH = 'v2/bot/message/push';
export const PATH_MULTICAST = 'v2/bot/message/multicast';
