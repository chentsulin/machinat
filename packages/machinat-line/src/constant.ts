export const LINE = 'line';

export const CHANNEL_API_CALL_GETTER = Symbol(
  'line.segment.channel_api_call_getter'
);
export const BULK_API_CALL_GETTER = Symbol('line.segment.bulk_api_call_getter');

export const PATH_REPLY = 'v2/bot/message/reply';
export const PATH_PUSH = 'v2/bot/message/push';
export const PATH_MULTICAST = 'v2/bot/message/multicast';
export const PATH_RICHMENU = 'v2/bot/richmenu';
