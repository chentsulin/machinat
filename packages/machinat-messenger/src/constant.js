// @flow
export const MESSENGER = 'messenger';
export const MESSENGER_NAMESPACE = 'Messenger';
export const MESSENGER_NATIVE_TYPE = Symbol.for('machinat.messenger.native');

export const ATTACHED_FILE_DATA = Symbol(
  'machinat.messenger.job.attached_file_data'
);

export const ATTACHED_FILE_INFO = Symbol(
  'machinat.messenger.job.attached_file_info'
);

export const ENTRY_MESSAGES = 'me/messages';

export const ENTRY_PASS_THREAD_CONTROL = 'me/pass_thread_control';
export const ENTRY_REQUEST_THREAD_CONTROL = 'me/request_thread_control';
export const ENTRY_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const ENTRY_MESSAGE_CREATIVES = 'me/message_creatives';
export const ENTRY_BROADCAST_MESSAGES = 'me/broadcast_messages';
export const ENTRY_CUSTOM_LABELS = 'me/custom_labels';
