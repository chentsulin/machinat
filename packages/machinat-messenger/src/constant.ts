export const MESSENGER = 'messenger';

export const ATTACHMENT_DATA = Symbol('messenger.segment.attachment_data');

export const ATTACHMENT_INFO = Symbol('messenger.segment.attachment_info');

export const ASSET_TAG = Symbol.for('messenger.segment.asset_tag');
export const ENTRY_PATH = Symbol.for('messenger.segment.entry');

export const PATH_MESSAGES = 'me/messages';

export const PATH_PASS_THREAD_CONTROL = 'me/pass_thread_control';
export const PATH_REQUEST_THREAD_CONTROL = 'me/request_thread_control';
export const PATH_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const PATH_MESSAGE_ATTACHMENTS = 'me/message_attachments';
export const PATH_MESSENGER_PROFILE = 'me/messenger_profile';
export const PATH_PERSONAS = 'me/personas';
