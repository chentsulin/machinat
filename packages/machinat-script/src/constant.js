// @flow
export const MACHINAT_SCRIPT_TYPE = Symbol.for('machinat.component.script');

export const SCRIPT_STATE_KEY = '$$machinat:script';
