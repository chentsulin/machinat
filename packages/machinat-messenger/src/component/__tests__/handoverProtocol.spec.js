import Machinat from 'machinat';

import { MESSENGER_NATIVE_TYPE } from '../../constant';
import {
  PassThreadControl,
  RequestThreadControl,
  TakeThreadContorl,
} from '../handoverProtocol';
import renderHelper from './renderHelper';

const render = renderHelper(null);

it.each([PassThreadControl, RequestThreadControl, TakeThreadContorl])(
  '%p is valid root Component',
  ThreadControl => {
    expect(typeof ThreadControl).toBe('function');
    expect(ThreadControl.$$native).toBe(MESSENGER_NATIVE_TYPE);
    expect(ThreadControl.$$namespace).toBe('Messenger');
  }
);

it.each(
  [
    <PassThreadControl appId="Legolas" metadata="you have my bow" />,
    <RequestThreadControl metadata="give me the ring" />,
    <TakeThreadContorl metadata="my precious" />,
  ].map(ele => [ele.type.name, ele])
)('%s match snapshot', async (_, threadControlElement) => {
  await expect(render(threadControlElement)).resolves.toMatchSnapshot();
});
