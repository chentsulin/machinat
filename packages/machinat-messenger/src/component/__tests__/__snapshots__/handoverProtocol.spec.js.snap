// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassThreadControl match snapshot 1`] = `
Array [
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "appId": "Legolas",
        "metadata": "you have my bow",
      },
      "type": [Function],
    },
    "path": "$",
    "type": "unit",
    "value": Object {
      "metadata": "you have my bow",
      "target_app_id": "Legolas",
      Symbol(machinat.messenger.segment.entry): "me/pass_thread_control",
    },
  },
]
`;

exports[`RequestThreadControl match snapshot 1`] = `
Array [
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "metadata": "give me the ring",
      },
      "type": [Function],
    },
    "path": "$",
    "type": "unit",
    "value": Object {
      "metadata": "give me the ring",
      Symbol(machinat.messenger.segment.entry): "me/request_thread_control",
    },
  },
]
`;

exports[`TakeThreadContorl match snapshot 1`] = `
Array [
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "metadata": "my precious",
      },
      "type": [Function],
    },
    "path": "$",
    "type": "unit",
    "value": Object {
      "metadata": "my precious",
      Symbol(machinat.messenger.segment.entry): "me/take_thread_control",
    },
  },
]
`;
